app/models/target.rb

```diff
@@ -30,6 +30,8 @@ class Target < ApplicationRecord
 
   belongs_to :user, counter_cache: true
   belongs_to :topic
+  has_noticed_notifications model_name: 'Notification'
+  has_many :notifications, through: :user, dependent: :destroy
 
```

Comment:
I would move the `has_noticed_notifications` below the associations since it's related to the gem and not to something specific of the associations.

```suggestion
  has_many :notifications, through: :user, dependent: :destroy
  
  has_noticed_notifications model_name: 'Notification'
```